au4/src/trackedit/internal/au3/au3interaction.cpp

```diff
@@ -328,7 +335,15 @@ bool Au3Interaction::pasteFromClipboard(secs_t begin, TrackId destinationTrackId
             clipIdsBefore.insert(clip.key.clipId);
         }
 
-        dstWaveTrack->Paste(begin, *clipboard()->trackData()[i].track);
+        if (clipboard()->trackData()[i].track.get()->NChannels() == 1 && dstWaveTrack->NChannels() == 2) {
+            // When the source is mono, may paste its only channel
+            // repeatedly into a stereo track
+            const auto pastedTrack = std::static_pointer_cast<WaveTrack>(clipboard()->trackData()[i].track);
+            pastedTrack->MonoToStereo();
+            dstWaveTrack->Paste(begin, *pastedTrack);
+        } else {
+            dstWaveTrack->Paste(begin, *clipboard()->trackData()[i].track);
```

Comment:
Please try to use the `at` method to get the element to read, not the `[]` operator. The `at` method definitely returns only a constant reference, we clearly see that the element will not be changed, and there is a check for access boundaries.

And in this case I think it would be better to add a method something like `trackData(size_t i) const`